swift/apple/FirezoneKit/Sources/FirezoneKit/Models/Token.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  Token.swift
+//  (c) 2024 Firezone, Inc.
+//  LICENSE: Apache-2.0
+//
+//  Convenience wrapper for working with our auth token stored in the Keychain.
+
+import Foundation
+
+public struct Token: CustomStringConvertible {
+  private static let query: [CFString: Any] = [
+    kSecAttrLabel: "Firezone token",
+    kSecAttrAccount: "1",
+    kSecAttrService: AppInfoPlistConstants.appGroupId,
+    kSecAttrDescription: "Firezone access token",
+  ]
+
+  private var data: Data
+
+  public var description: String { String(data: data, encoding: .utf8)! }
+
+  public init(_ data: Data) {
+    self.data = data
+  }
+
+  public static func delete() async throws {
+    guard let tokenRef = await Keychain.shared.search(query: query)
```

Comment:
Is it good to use global state for all of these? Why not pass in a reference here? Then you could unit-test all this with a stub-keychain. Not sure if Apple provides something here?

Reply:
> Not sure if Apple provides something here?

This is a joke right? 😆 

But yes, I can update this to pass in a Keychain reference to set up us for better testing later.

Tests are going to a be a big can of worms. Apple doesn't seem very keen on making Network Extensions testable:

https://developer.apple.com/forums/thread/749579

Reply:
Well, unit-testing functions like this is hopefully possible :)

Can one create a keychain just for the test?

Reply:
> Can one create a keychain just for the test?

I'm not sure how useful this will be (if it's even possible) because the Keychain is so deeply tied into Apple's security subsystems. We use the System keychain for example to share data between between the GUI and tunnel process. On iOS, the keychain behaves differently in sometimes subtle ways, so we'll need to think a bit more about how to go about testing it.

Reply:
I think we could probably get an integration-level test working on the macOS GitHub runner after much effort that's able to live-test these system interfaces

Reply:
The tests I was thinking of would treat the keychain (or some protocol we define for it) as a store that you can read from and write to. Worst case, we'd have to define our own `Protocol` that gets implemented for `Keychain` and some stub to ensure our way of reading and writing to it is consistent. Really, what we'd then be testing is that the encoding and decoding works correctly.

We don't need to test that Apple's keychain works as advertised so we don't need an actual keychain in the tests.

Reply:
Yeah, agreed.

But I do want to emphasize how many sharp edges the underlying Keychain API here has.

Leave a dict key of the add/update call? `EXC_BAD_ACCESS` and crash. Supply an extra key in the search call? Same.

I'm sure there are edge cases lurking here that we may be susceptible too but just haven't hit the right Keychain state for them to pop up.

I.e. the encoding and decoding is the easy part because it's compile-time checked. I think these ObjC-era APIs are just a big headache. I guess we're spoled.